libs/checkpoint-postgres/langgraph/checkpoint/postgres/__init__.py

```diff
@@ -379,10 +386,17 @@ def _cursor(self, *, pipeline: bool = False) -> Iterator[Cursor[DictRow]]:
             elif pipeline:
                 # a connection not in pipeline mode can only be used by one
                 # thread/coroutine at a time, so we acquire a lock
-                with self.lock, conn.pipeline(), conn.cursor(
-                    binary=True, row_factory=dict_row
-                ) as cur:
-                    yield cur
+                try:
```

Comment:
We can't do this without trying every time?

Reply:
good point, will update